src/v/iceberg/metadata_query.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+#include "base/seastarx.h"
+#include "iceberg/manifest_io.h"
+#include "iceberg/table_metadata.h"
+
+#include <seastar/core/future.hh>
+
+namespace iceberg {
+
+/**
+ * Enum class allowing changing an expected type returned from query
+ */
+enum class result_type { snapshot, manifest_file, manifest };
```

Comment:
I wasn't sure if we wanted data_file in here too, or is the idea that it's simple once we have the manifest in hand?

Reply:
Oh I think i get it -- these are corresponding to IO required to fetch them, and we won't fetch the data files

Reply:
exactly, initially i was thinking about data files as well but they can be get from the manifest.